crates/protocol/protocol/src/block.rs

```diff
@@ -144,8 +143,8 @@ impl L2BlockInfo {
     }
 
     /// Constructs an [L2BlockInfo] from a given OP [Block] and [ChainGenesis].
-    pub fn from_block_and_genesis(
-        block: &OpBlock,
+    pub fn from_block_and_genesis<T: Typed2718>(
+        block: &Block<T>,
```

Comment:
So this will still error because the `T` type doesn't implement `as_deposit()`.

Since the `TransactionResponse` of the `Optimism` network trait is `op_alloy_rpc_types::Transaction` which only internally contains `OpTxEnvelope`, there's no `as_deposit` available.

I _think_ we can get around this by restricting `T` to `AsRef<OpTxEnvelope>` and then implement it on `OpTxEnvelope`.

I've opened a PR (https://github.com/alloy-rs/op-alloy/pull/464) in `op-alloy` to do this cc @mattsse.